src/raftstore/store/peer.rs

```diff
@@ -813,16 +822,8 @@ impl Peer {
         if self.should_read_local(&req) {
             metrics.local_read += 1;
 
-            // Execute ready-only commands immediately in leader when doing local reads.
-            let mut ctx = ExecContext::new(self, 0, 0, &req);
-            let (mut resp, _) = self.exec_raft_cmd(&mut ctx).unwrap_or_else(|e| {
-                error!("{} execute raft command err: {:?}", self.tag, e);
-                (cmd_resp::new_error(e), None)
-            });
-
-            cmd_resp::bind_uuid(&mut resp, meta.uuid);
-            cmd_resp::bind_term(&mut resp, self.term());
-            cb(resp);
+            let read_task = ApplyTask::read(self.region_id, self.term(), req, cb);
```

Comment:
No, for example, a client tries to write (k1, v1), and then it meets timeout error. It retries to see if (k1, v1) has been written successfully. If a snapshot is generated directly here, client may get None but (k1, v1) may be actually written later.